src/Resource.php

```diff
@@ -80,8 +80,16 @@ public function toArray()
             $array['relationships'] = $relationships;
         }
 
+        $links = [];
         if (! empty($this->links)) {
-            $array['links'] = $this->links;
+            $links = $this->links;
+        }
+        $serializerLinks = $this->serializer->getLinks($this->data);
+        if (! empty($serializerLinks)) {
+            $links = array_merge($serializerLinks, $links);
+        }
```

Comment:
Ah yes, didn't see the test and misread the order of `array_merge` arguments – the current behaviour is fine.